pytorch_lightning/core/optimizer.py

```diff
@@ -185,10 +162,3 @@ def closure_dis():
         assert trainer is not None
         with trainer.profiler.profile(profiler_action):
             trainer.accelerator.optimizer_step(self._optimizer, self._optimizer_idx, closure, **kwargs)
-
-    def __repr__(self) -> str:
```

Comment:
I think it is still useful to know you aren't using our own optimizer, but the lightning optimizer one. This is definitely useful for debugging purposes.

Reply:
The test below shows it. 
You can try:

```py
    opt = LightningOptimizer(torch.optim.Adam(torch.nn.Linear(1, 1).parameters()))
    print(opt)
```
    
prints on master:
```py
LightningAdam(groups=[{'amsgrad': False, 'betas': (0.9, 0.999), 'eps': 1e-08, 'lr': 0.001, 'weight_decay': 0}])
```

prints here:
```
LightningAdam (
Parameter Group 0
    amsgrad: False
    betas: (0.9, 0.999)
    eps: 1e-08
    lr: 0.001
    weight_decay: 0
)
```

And if you did

```py
    opt = torch.optim.Adam(torch.nn.Linear(1, 1).parameters())
    print(opt)
```

you get 

```
Adam (
Parameter Group 0
    amsgrad: False
    betas: (0.9, 0.999)
    eps: 1e-08
    lr: 0.001
    weight_decay: 0
)
```

Reply:
I removed it because this is supposed to be a tiny wrapper for us to hook into our accelerators.

By writing our own `__repr__`, we override the one available in the optimizer passed.

Perhaps the user has its own custom Optimizer, with its own custom `__repr__`. We want to use it in that case.

If you think that the PyTorch optimizers should have a more readable `__repr__`, this is something that should be raised and implemented in PyTorch directly. This wrapper should not make choices for the user.

Reply:
Opened https://github.com/pytorch/pytorch/issues/67587.

Resolving this